firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -174,7 +174,16 @@ public void setFirestoreSettings(@NonNull FirebaseFirestoreSettings settings) {
   }
 
   private void ensureClientConfigured() {
-    if (client == null) {
+    // We do not bother to synchronized on client as we do not touch it if it is already set.
+    if (client != null) {
+      return;
+    }
+    synchronized (settings) {
```

Comment:
Since you're trying to prevent `this.client` from being created a second time, synchronizing on `this` might make sense... though I don't think it matters given that there aren't any other sync blocks in this class.

(You might want to protect `setFirestoreSettings()` too, though it's docs claim that 'This method can only be called before calling any other methods on this object', so I we'd only get in trouble if the user misused the api. Might still be worth it?)

Reply:
Synchronize `this` on public object instances is generally not a good idea because nothing prevents the caller from also be synchronizing on `this`.